aldryn_newsblog/cms_wizards.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from django.utils.translation import ugettext_lazy as _
+from django import forms
+
+from cms.api import add_plugin
+from cms.utils import permissions
+from cms.wizards.wizard_pool import wizard_pool
+from cms.wizards.wizard_base import Wizard
+from cms.wizards.forms import BaseFormMixin
+
+from parler.forms import TranslatableModelForm
+
+from .cms_appconfig import NewsBlogConfig
+from .models import Article
+
+
+class NewsBlogArticleWizard(Wizard):
+
+    def user_has_add_permission(self, user, **kwargs):
+        """
+        Return True if the current user has permission to add an article.
+        :param user: The current user
+        :param kwargs: Ignored here
+        :return: True if user has add permission, else False
+        """
+        # No one can create an Article, if there is no app_config yet.
+        num_configs = NewsBlogConfig.objects.count()
+        if not num_configs:
+            return False
+
+        # Ensure user has permission to create articles.
+        if user.is_superuser or user.has_perm("aldryn_newsblog.add_article"):
+            return True
+
+        # By default, no permission.
+        return False
+
+
+class CreateNewsBlogArticleForm(BaseFormMixin, TranslatableModelForm):
+    """
+    The ModelForm for the NewsBlog article wizard. Note that Article has a
+    number of translated fields that we need to access, so, we use
+    TranslatableModelForm
+    """
+
+    content = forms.CharField(
+        label="Content", help_text=_("Optional. If provided, will be added to "
+                                     "the main body of the article."),
+        required=False, widget=forms.Textarea())
+
+    class Meta:
+        model = Article
+        fields = ['title', 'app_config', 'is_published', 'lead_in', 'content', ]
+
+    def __init__(self, **kwargs):
+        super(CreateNewsBlogArticleForm, self).__init__(**kwargs)
+
+        # If there's only 1 (or zero) app_configs, don't bother show the
+        # app_config choice field, we'll choose the option for the user.
+        app_configs = NewsBlogConfig.objects.all()
+        if app_configs.count() < 2:
+            self.fields['app_config'].widget = forms.HiddenInput()
```

Comment:
Good, but we probably have to make a markup flexible enough for this or at least to communicate the fact that this field might be not visible.